src/components/international.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import Polyglot from 'node-polyglot';
+import { Provider } from '../context';
+import smartT from '../smart_t';
+
+const warn = (...args) => console.error(...args);
+
+const createInstance = ({ type: locale, dictionary: phrases }) => {
+  const polyglot = new Polyglot({ warn, locale, phrases });
+  return (...args) => smartT(polyglot, ...args);
```

Comment:
Not sure if we _always_ need `smartT`. I think it would be enough if we move `smartT`s functionality into `T` component and return polyglot `t` here.

`t` -> returns string
`<T>` -> returns reacty stuff

Reply:
i would like to keep a simple function. i think performance wise it's about the same, it nopes out of parsing/rendering if no react components were passed